ocsp/responder.go

```diff
@@ -49,6 +51,59 @@ func (src InMemorySource) Response(request *ocsp.Request) (response []byte, pres
 	return
 }
 
+// DBSource represnts a source of OCSP responses backed by certdb
+type DBSource struct {
+	Accessor certdb.Accessor
+}
+
+// NewDBSource creates a new DBSource type with associated dbAccessor
+func NewDBSource(dbAccessor certdb.Accessor) Source {
+	return DBSource{
+		Accessor: dbAccessor,
+	}
+}
+
+// Response implements cfssl.ocsp.responder.Source, returning the OCSP response
+// with the expiration date furthest in the future
+func (src DBSource) Response(req *ocsp.Request) ([]byte, bool) {
+	if req == nil {
+		return nil, false
+	}
+
+	// Extract the AKI string from req.IssuerKeyHash
+	aki := hex.EncodeToString(req.IssuerKeyHash)
+
+	sn := req.SerialNumber
+	if sn == nil {
+		return nil, false
+	}
+	strSN := sn.String()
+
+	if src.Accessor == nil {
+		log.Errorf("No DB Accessor")
+		return nil, false
+	}
+	records, err := src.Accessor.GetOCSP(strSN, aki)
+
+	// Log on errors obtaining OCSP response
```

Comment:
Should anything else happen when there's an error? Should the function continue? It would be useful to document that in a comment here.